File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const hashMap = new Map();
    
    for(let value of participant){
        if(hashMap.has(value)){
            hashMap.set(value, hashMap.get(value)+1);
        }
        else{
            hashMap.set(value,1);
        }
    }
    
    
    for(let value of completion){
        if(hashMap.get(value)>0){
            hashMap.set(value,hashMap.get(value)-1);
        }
    }
    
    for(let [key,value] of hashMap){
        if(value===1) return key;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):        
    hash_dict = {}
    
    for p in participant:
        hash_dict[p] = hash_dict.get(p,0)+1
    
    for c in completion:
        hash_dict[c] -= 1
        
    for key, value in hash_dict.items():
        if value>0:
            return key
